server/app.js

```diff
@@ -1,43 +1,61 @@
-const path = require('path')
-const express = require('express')
-const morgan = require('morgan')
-const app = express()
-module.exports = app
+const path = require('path');
+const express = require('express');
+const morgan = require('morgan');
+const User = require('./db/models/User');
+const app = express();
+module.exports = { app, requireToken };
 
 // logging middleware
-app.use(morgan('dev'))
+app.use(morgan('dev'));
 
 // body parsing middleware
-app.use(express.json())
+app.use(express.json());
+
+// token checking middleware (used in /api)
+async function requireToken(req, res, next) {
```

Comment:
define async middleware function that checks token. it is exported along with ```app``` near the top of the file